File: content/examples/other/leetcode/problems/counting-bits/example-2/solution.py
def count_bits(n):
    out = [0] * (n + 1)
    for i in range(1, n + 1):
        out[i] = out[i >> 1] + (i & 1)
    return out


if __name__ == "__main__":
    print(count_bits(5))

File: content/examples/other/leetcode/problems/counting-bits/example-2/Solution.java
import java.util.Arrays;

public class Solution {
  static int[] countBits(int n) {
    int[] out = new int[n + 1];
    for (int i = 1; i <= n; i += 1) {
      out[i] = out[i >> 1] + (i & 1);
    }
    return out;
  }

  public static void main(String[] args) {
    System.out.println(Arrays.toString(countBits(5)));
  }
}
